### PR TITLE
Add support for private registry

### LINKED ISSUE #245: DeployableByOLM fails in k8s use cases if images are private
## Bug Description
If the bundle image and/or index image are in a private registry, the DeployableByOLM check will fail to successfully execute.

## Version and Command Invocation
Preflight version is bf7386394b9db59ca9020b0d5363ac75

## Steps to Reproduce:
I've only experienced this while running Preflight in k8s as a job, but it may be reproducible otherwise. The basic steps look something like this:

1) Upload an index image and a bundle image in a private repo
2) Run preflight in k8s as a Job, mounting secrets such as the image's registry creds and a kubeconfig to the proper places.
3) Wait for preflight to complete
4) Review logs, see DeployableByOLM listed as failed

## Expected Result
DeployableByOLM should pass this check.

## Actual Result
DeployableByOLM is returned as an errored check (not a failed check) which implies that something went wrong in the execution.

## Additional Context
With `PFLT_LOGLEVEL=trace`, there still isn't much context as to what happened. However, given that the index image is private, I'm thinking that the problem is that the CatalogSource that's created by this check cannot pull the index image.

In addition, the namespace that's used for this check (which, I believe, is introspected from the package name) likely doesn't have pull secrets necessary to try and run this operator, so I would expect the check would fail beacuse the CSV may never reach "Ready". 

The install namespace is dynamically spun up, which makes presenting credentials tough. In theory, presenting credentials to the openshift-marketplace namespace (where we create the catalogsource) might be easier, but still means we're frequently creating credentials in this namespace.

The pipelines don't really have this challenge because they have access to the bundle itself, and dynamically create an index and push it to a known location for testing.

### PR DESCRIPTION
**Pull request by samira-barouti**
This PR enables the preflight to work with the index images that are in private
registries. It introduces a new mandatory DOCKERCONFIG environment variable
which must point to the docker config file.

Fixes #245

### CODE REVIEW

File: certification/internal/policy/operator/deployable_by_olm.go [hunk 1]
```diff
@@ -75,10 +76,6 @@ func (p *DeployableByOlmCheck) operatorMetadata(bundleRef certification.ImageRef
 	}
 
 	catalogImage := viper.GetString(indexImageKey)
-	if len(catalogImage) == 0 {
```
Comment (acornett21) [inline certification/internal/policy/operator/deployable_by_olm.go:78]: Is this validation no longer needed? Just want to make sure it wasn't removed on accident, as I don't see any validation for the new secrete code.

Comment (samira-barouti) [reply certification/internal/policy/operator/deployable_by_olm.go:78]: I moved this to check_operator to ensure the check is failing fast.

**Review by bcrochet — COMMENTED** [review]
What happens if there is no need for the secret? There are certainly going to be instances where this is not necessary. It seems to force it no matter what.

File: certification/internal/policy/operator/deployable_by_olm.go (hunk 1 above)
Comment (acornett21) [reply certification/internal/policy/operator/deployable_by_olm.go:78]: That makes sense, thanks for clarifying!

**Comment by samira-barouti** [comment]
> What happens if there is no need for the secret? There are certainly going to be instances where this is not necessary. It seems to force it no matter what.

Just ran the preflight with a publicly-available index image and it is working fine. 
I also test it with multiple auth keys (one per repo) in the config.json and that is working well.

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**